### ISSUE TITLE
Incompatibility of Pint-Pandas==0.4 with pandas==2.1.0

### ISSUE THREAD

**Issue by aaaandi** [hgrecco/pint-pandas#206]
As stated in the release notes of pandas 2.1.0 ([Other API changes](https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#other-api-changes)), the `arrays.PandasArray` has been renamed `NumpyExtensionArray` and the attached dtype name changed from `PandasDtype` to `NumpyEADtype`.

As a result, I get the following error when importing Pint-pandas:

```
from .pint_array import PintArray, PintType
np.complex128: pd.core.dtypes.dtypes.PandasDtype("complex128"),
E   AttributeError: module 'pandas.core.dtypes.dtypes' has no attribute 'PandasDtype'
```

Versions are Pint-Pandas==0.4 and pandas==2.1.0.

**Comment by MichaelTiemannOSC** [comment]
Should be fixed in Pint-Pandas 0.5, released a few days ago.

**Comment by aaaandi** [comment]
Oh, perfect. Thanks a lot!
